lib/questionnaires/model/item/answer/src/answer_model.dart

```diff
@@ -124,6 +124,9 @@ abstract class AnswerModel<I, V> extends ResponseNode {
   /// Returns whether this question is unanswered.
   bool get isEmpty;
 
+  /// Is there a dialog shown on screen?
+  bool isDialogShown = false;
+
```

Comment:
Thank you for the addition.
I like the idea of adding the property, but I have several concerns:

1. I think we can make this more general, for example, `isUserInteractionAllowed`. This variable will hold the logic of whether the user is allowed to interact with the question, e.g. a dialog is shown, has running tasks, etc.
1. IMO, the property should belong to `QuestionItemModel` instead, since the property is kinda more related to the question itself.

Reply:
Thank you for the review. 

For the `isUserInteractionAllowed` property will it be added in the theme in smart_forms_flutter_sdk? Right now we use the `isDialogShown` property provided by faiadashu in the join_triage pageValidation callback added [here](https://git.allm.net/fhir-forms/smart-forms-flutter-sdk/-/merge_requests/32/diffs#3c8414eb1330ba3f9d36ec1ce59830c5dd688fa4)

Reply:
Moved the `isDialogShown` flag to `QuestionItemModel`

Reply:
Sorry I was not clear.  To me,`isDialogShown` is really specific, so I think it would be good to rename it to be more general.
My suggestion is `isUserInteractionAllowed`, so if ever we have other behavior that makes the user not allowed to do something, then we can just reuse this variable.